src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/Marshalling/ICustomNativeTypeMarshallingStrategy.cs

```diff
@@ -503,35 +500,34 @@ public IEnumerable<StatementSyntax> GenerateMarshalStatements(TypePositionInfo i
             }
 
             if (!CanPinMarshaller(info, context))
-            {
-                // <nativeIdentifier> = <marshalerIdentifier>.Value;
-                yield return ExpressionStatement(
-                    AssignmentExpression(
-                        SyntaxKind.SimpleAssignmentExpression,
-                        IdentifierName(context.GetIdentifiers(info).native),
-                        MemberAccessExpression(SyntaxKind.SimpleMemberAccessExpression,
-                            IdentifierName(subContext.GetIdentifiers(info).native),
-                            IdentifierName(ManualTypeMarshallingHelper.ValuePropertyName))));
-            }
+                yield return GenerateNativeAssignmentFromValueProperty(info, context, subContext);
+        }
+
+        private static StatementSyntax GenerateNativeAssignmentFromValueProperty(TypePositionInfo info, StubCodeContext context, CustomNativeTypeWithValuePropertyStubContext subContext)
+        {
+            // <nativeIdentifier> = <marshalerIdentifier>.Value;
+            return ExpressionStatement(
+                AssignmentExpression(
+                    SyntaxKind.SimpleAssignmentExpression,
+                    IdentifierName(context.GetIdentifiers(info).native),
+                    MemberAccessExpression(SyntaxKind.SimpleMemberAccessExpression,
+                        IdentifierName(subContext.GetIdentifiers(info).native),
+                        IdentifierName(ManualTypeMarshallingHelper.ValuePropertyName))));
         }
 
         public IEnumerable<StatementSyntax> GeneratePinStatements(TypePositionInfo info, StubCodeContext context)
         {
-            // fixed (<_nativeTypeSyntax> <nativeIdentifier> = &<marshalerIdentifier>)
+            // fixed (<_nativeTypeSyntax> <ignoredIdentifier> = &<marshalerIdentifier>)
+            //  <assignment to Value property>
             var subContext = new CustomNativeTypeWithValuePropertyStubContext(context);
             yield return FixedStatement(
                 VariableDeclaration(
                 _valuePropertyType,
                 SingletonSeparatedList(
-                    VariableDeclarator(context.GetIdentifiers(info).native)
+                    VariableDeclarator(Identifier(context.GetAdditionalIdentifier(info, "ignored")))
```

Comment:
Instead of `"ignored"` do we want to use `_`? I assume the later is the canonical way to do this in C#. Is there a reason we might want to be explicit here?

Reply:
We can't use `_` as it's still a valid identifier name. As a result, if we have two parameters that both hit this path, we get errors about `_` having already been declared. 😢

Reply:
boo.